registry/handlers/images.go

```diff
@@ -196,6 +196,7 @@ func (imh *imageManifestHandler) convertSchema2Manifest(schema2Manifest *schema2
 		imh.Errors = append(imh.Errors, v2.ErrorCodeManifestInvalid.WithDetail(err))
 		return nil, err
 	}
+	imh.Digest = digest.FromBytes(manifest.(*schema1.SignedManifest).Canonical)
```

Comment:
I did test this with 1.6 and 1.7 and can confirm looking at the code it
does not include the signatures.

On Tue, Feb 9, 2016, 6:33 PM Aaron Lehmann <notifications@github.com> wrote:

> In registry/handlers/images.go
> <https://github.com/docker/distribution/pull/1445#discussion_r52408191>:
>
> > @@ -196,6 +196,7 @@ func (imh *imageManifestHandler) convertSchema2Manifest(schema2Manifest *schema2
> >  		imh.Errors = append(imh.Errors, v2.ErrorCodeManifestInvalid.WithDetail(err))
> >  		return nil, err
> >  	}
> > +	imh.Digest = digest.FromBytes(manifest.(*schema1.SignedManifest).Canonical)
>
> Are you sure Canonical is correct? This is the of the manifest without
> any attached signatures.
>
> The HTTP response does have attached signatures, so I would expect we'd
> want to digest the complete manifest returned by Payload.
>
> Of course, it depends whether 1.6 / 1.7 include the signatures when
> performing this digest calculation. I don't know whether they do or not.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/distribution/pull/1445/files#r52408191>.
>